opencog/util/cluster.c

```diff
@@ -2922,10 +2922,7 @@ If a memory error occurs, pclcluster returns NULL.
   }
 
   /* Free temporarily allocated space */
-  free(data[0]);
-  free(mask[0]);
-  free(data);
-  free(mask);
+  freedatamask(nelements, data, mask);
```

Comment:
We unfortunately can't test since no more code is using that function, but that seems correct nevertheless.